scipy/ndimage/src/nd_image.c

```diff
@@ -190,6 +194,9 @@ static PyObject *Py_Correlate1D(PyObject *obj, PyObject *args)
 
     NI_Correlate1D(input, weights, axis, output, (NI_ExtendMode)mode, cval,
                    origin);
+    #if NPY_API_VERSION >= 0x0000000c
+        PyArray_ResolveWritebackIfCopy(output);
+    #endif
```

Comment:
This C preprocessor if / endif block is used quite a lot in this PR in different places. Could it perhaps be helpful from a maintenance / code duplication standpoint to abstract the preprocessor code to a single macro declaration at the top of the C module, or perhaps even better in one of the associated header files that is included?

Reply:
Yes, `HAVE_WRITEBACKIFCOPY` please